src/filesystem_helper.cpp

```diff
@@ -371,12 +371,13 @@ bool create_directories(const path & p)
     } else {
       p_built = *it;
     }
-    if (!p_built.exists()) {
```

Comment:
Yeah, agreed with that.  This entire thing is inherently racy; if there are other processes or threads messing with the directory structure while this function is executing, then there are several other things that can fail as well.  I'm fine with adding the check at the end so that if we get EEXIST, we consider that a success.

Reply:
fixed in [d5a9c1a](https://github.com/ros2/rcpputils/pull/162/commits/d5a9c1a944ea9147c2d2ca8fd87b96ff5bcb03e6)